rfc/0002_scope_manager.md

```diff
@@ -0,0 +1,61 @@
+# 0002: SpanManager
+
+**Current State:** Draft  
+**Author:** [carlosalberto](https://github.com/carlosalberto)
+
+In the OpenTracing specification, under the "Optional API Elements" section, it is mentioned languages may choose to provide utilities to pass an active **Span** around a single process.
+
+Upon many iterations and feedback from several contributors, the Java 0.31 API defined the new **Scope Manager** concept, which is a simple and explicit way to manage the active **Span** at a given call-context point. This document intends to standardize this concept, so other supported languages and platforms can leverage it, with their respective semantical differences.
+
+# Technical background
+
+For any thread, at most one **Span** may be "active". Of course there may be many other **Spans** involved with the thread which are (a) started, (b) not finished, and yet (c) not "active": perhaps they are waiting for I/O, blocked on a child **Span**, or otherwise off of the critical path.
+ 
+It's inconvenient to pass an active **Span** from function to function manually, so OpenTracing should require that every **Tracer** contains a **Scope Manager** that grants access to the active **Span** through a container, called **Scope**.
+
+# Specification Changes
+
+New `ScopeManager` and `Scope` interfaces are added to the specification, and the `Tracer` interface is extended to support creation of `Span`s that are automatically set as the **active** one for the current context.
+
+## ScopeManager
+
+The `ScopeManager` interface allows setting the active `Span` in a call-context storage section, and has the following members:
+
+* **activate**, capability to set the specified `Span` as the active one for the current call-context, returning a `Scope` containing it. A parameter (still to decide whether it will be optional or required) will mark whether the returned `Scope` should, upon deactivation, **finish** the contained `Span`.
```

Comment:
My argument for keeping the parameter required here is that the `activate` method gets an **existing** span so it should also be up to the caller whether or not the span should be finished when the scope is closed.

Reply:
Definitely, not as 'obvious' as compared to **start active span**. I will leave it as it is for now.